app/channel.go

```diff
@@ -1370,11 +1370,37 @@ func (a *App) MoveChannel(team *model.Team, channel *model.Channel) *model.AppEr
 		}
 	}
 
-	// Change the Team ID of the channel.
+	// keep instance of the previous team
+	var previousTeam *model.Team
+	if result := <-a.Srv.Store.Team().Get(channel.TeamId); result.Err != nil {
+		return result.Err
+	} else {
+		previousTeam = result.Data.(*model.Team)
+	}
 	channel.TeamId = team.Id
 	if result := <-a.Srv.Store.Channel().Update(channel); result.Err != nil {
 		return result.Err
 	}
+	a.postChannelMoveMessage(user, channel, previousTeam)
+
+	return nil
+}
+
+func (a *App) postChannelMoveMessage(user *model.User, channel *model.Channel, previousTeam *model.Team) *model.AppError {
+
+	post := &model.Post{
+		ChannelId: channel.Id,
+		Message:   fmt.Sprintf(utils.T("api.team.move_channel.success"), previousTeam.Name),
+		Type:      model.POST_MOVE_CHANNEL,
+		UserId:    user.Id,
```

Comment:
What does this parameter get set to for system messages? It seems like this message might be more appropriate if it isn't attributed to a particular user. If we want to track who did the move, we should do so in an audit table. I only bother to bring this up because setting this parameter requires a change to the CLI, which could break existing scripts that users have written against Mattermost.

Reply:
That was an agreed upon concern. The original PR #8014 has a large discussion around it. tldr; was since UserId is required by model.Post the --username would be added until a "matterbot" type user could be implemented.

Reply:
Good enough for me. I wasn't aware of the previous discussion. Thanks for clarifying.